spec/app_spec.rb

```diff
@@ -0,0 +1,116 @@
+require 'spec_helper'
+require 'rack/test'
+require 'omniauth-myusa'
+require 'stub_classes'
+
+describe 'ComplianceViewer' do
+  include Rack::Test::Methods
+
+  class StubComplianceData
+  end
+
+  def app
+    ComplianceViewer.new! StubComplianceData.new
```

Comment:
Took me a while to figure that one out. :) Sinatra redefines `new` to include the Rack middleware. Before doing that, it aliases the existing functionality to `new!`. I do that so I can get access an actual `ComplianceViewer` instance.

https://github.com/sinatra/sinatra/blob/v1.3.3/lib/sinatra/base.rb#L1369-1370